pkg/core/ca/issuer/issuer.go

```diff
@@ -42,18 +42,22 @@ func NewWorkloadCert(ca util_tls.KeyPair, mesh string, workload string, certOpts
 	if err != nil {
 		return nil, errors.Wrap(err, "failed to generate a private key")
 	}
-	workloadCert, err := newWorkloadCert(caPrivateKey, caCert, mesh, workload, workloadKey.Public(), certOpts...)
+	workloadCert, err := newWorkloadCert(caPrivateKey, caCert, mesh, services, workloadKey.Public(), certOpts...)
 	if err != nil {
 		return nil, errors.Wrap(err, "failed to generate X509 certificate")
 	}
 	return util_tls.ToKeyPair(workloadKey, workloadCert)
 }
 
-func newWorkloadCert(signer crypto.PrivateKey, parent *x509.Certificate, trustDomain string, workload string, publicKey crypto.PublicKey, certOpts ...CertOptsFn) ([]byte, error) {
-	spiffeID := &url.URL{
-		Scheme: "spiffe",
-		Host:   trustDomain,
-		Path:   workload,
+func newWorkloadCert(signer crypto.PrivateKey, parent *x509.Certificate, trustDomain string, services []string, publicKey crypto.PublicKey, certOpts ...CertOptsFn) ([]byte, error) {
```

Comment:
That's a lot of arguments to pass. Maybe a structure could be better.

Reply:
Ok, I applied some refactoring in this area. It doesn't make sense to carry all of those args across 2 methods.